planner/core/logical_plan_builder.go

```diff
@@ -4140,6 +4141,31 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 		us.SetChildren(ds)
 		result = us
 	}
+	// If a table is a cache table, it is judged whether it satisfies the conditions of read cache.
+	if tableInfo.TableCacheStatusType == model.TableCacheStatusEnable {
+		cachedTable := tbl.(table.CachedTable)
+		txn, err := b.ctx.Txn(true)
+		if err != nil {
+			return nil, err
+		}
+		// Use the txn of the transaction to determine whether the cache can be read.
+		// About read lock and read condition feature. will add in the next pr.
+		cond := cachedTable.IsReadFromCache(txn.StartTS())
+		b.ctx.GetSessionVars().StmtCtx.StoreCacheTableReadCondition(tbl.Meta().ID, cond)
+		if cond {
+			us := LogicalUnionScan{handleCols: handleCols}.Init(b.ctx, b.getSelectOffset())
+			us.SetChildren(ds)
+			result = us
+		} else {
+			go func() {
+				err := cachedTable.UpdateLockForRead(b.ctx, txn.StartTS())
+				if err != nil {
+					log.Warn("Update Lock Info Error")
```

Comment:
If lock failed, will it cause some, will executor return a wrong result?

Reply:
if lock failed. next read will read normal table. i think its not generate wrong.